### ISSUE TITLE
error message when executing :VimwikiIndex, Undefined variable: vimhome

### ISSUE THREAD

**Issue by jean-francois-labbe** [vimwiki/vimwiki#1345]
Prior to submitting a new issue make sure to complete these steps:

- [x] Include the VimWiki settings from your `.vimrc`
- [x] Include the syntax you are using (default / Markdown / MediaWiki) 
- [x] Provide a detailed description of the problem including **steps to reproduce the issue**.
- [x] Include the output of `:VimwikiShowVersion`.

`:VimwikiShowVersion` (is working only after `<leader>ww`)
```
Version: 2023.05.12
Os: Mac
Vim: 801
Branch: dev
Revision: cd9cfd2
Date: 2023-05-12 22:05:10 -0400
```

**neovim v0.10.0-dev-2601**

I've just added vimwiki as a dependency with Lazy plugin loaded.

Facing this error message when executing `:VimwikiIndex`
```
Error detected while processing function tar#Vimuntar:
line   20:
E117: Unknown function: vimball#VimballHome
line   25:
E121: Undefined variable: vimhome
E116: Invalid arguments for function simplify
line   53:
E117: Unknown function: vimball#Decompress
line   65:
E121: Undefined variable: vimhome
E116: Invalid arguments for function simplify
```

Lazy infos:
```
    ● vimwiki 6.93ms  start
        dir    /Users/jeff/.local/share/nvim/lazy/vimwiki
        url    https://github.com/vimwiki/vimwiki
        branch dev
        commit cd9cfd2
        readme README.md
        help   |vimwiki-emoji.txt|
        help   |vimwiki.txt|
```

vimwiki Lazy configuration

```lua
  {
    "vimwiki/vimwiki",
    priority = 1000,
    branch = "dev",
    lazy = false,
    -- dependencies = {'michal-h21/vim-zettel'},
    init = function()
      vim.g.vimwiki_global_ext = 0
      vim.g.vimwiki_list = {
        {
          path = "~/vimwiki/",
          syntax = "markdown",
          ext = ".md",
        },
      }
    end
  },

```

when executing `<Leader>ww` it opens the correct file and then I have a lot more commands starting with `:Vimwiki...`

**Comment by tinmarino** [comment]
Hi,

This seems more related to [Lazy](https://github.com/folke/lazy.nvim) or [VimBall](https://github.com/vim-scripts/Vimball/blob/master/autoload/vimball.vim#L476-L534) than to VimWiki.

Anyway, I guess the
```
E121: Undefined variable: vimhome
```
Is expecting variable vimhome so why do not you declare it.
```
let g:vimhome= $HOME."/.vim"
``` 
Or similar for nvim.

Closing as not related to vimwiki <= there is no vimhome variable here

**Comment by jean-francois-labbe** [comment]
Thanks for the answer, and sorry for the noise.
I asked here as this is only happening when launching the plugin.

**Comment by tinmarino** [comment]
Well actually thank you for the "noise", others may come here too. But I cannot (lazyly) reproduce.

Did you find a fix?

**Comment by tinmarino** [comment]
Roazon en force mec 💪
(j'y ai vécu 5 ans)

**Comment by jean-francois-labbe** [comment]
Yes I managed to find out the issue. It was due to "hrsh7th/nvim-cmp"
Here is the configuration that caused the issue
```lua
      local cmp = require("cmp")
      local cmp_select = { behavior = cmp.SelectBehavior.Select }
      local cmp_mappings = lsp.defaults.cmp_mappings({
        ["<CR>"] = cmp.mapping({
            c = cmp.mapping.confirm({ behavior = cmp.ConfirmBehavior.Replace, select = true }),
          }),
```
If I change `select = true` to `select = false` the issue goes away.
In the end I completely removed ` c = cmp.mapping.confirm({ behavior = cmp.ConfirmBehavior.Replace, select = true }),`
I copied setup from README without understanding it.

Now I can enjoy Vimwiki 👍

**Comment by jean-francois-labbe** [comment]
Roazon is nice 😃
